src/context/PlaylistProvider.jsx

```diff
@@ -0,0 +1,151 @@
+            .catch((err) => {
+                console.log(err);
+            });
+    };
+
+    const addPlaylist = (title, description) => {
+        if (!auth.isAuthenticated) {
+            console.log("User is not Authenticated");
+            return;
+        }
+        axios
+            .request({
+                method: "post",
+                url: "/api/user/playlists",
+                headers: { authorization: auth.authToken },
+                data: {
+                    playlist: {
+                        title,
+                        description,
+                    },
+                },
+            })
+            .then((res) => {
+                console.log(res.data.playlists);
+                setPlaylist([...res.data.playlists]);
+            })
+            .catch((err) => {
+                console.log(err);
+            });
+    };
+
+    const removePlaylist = (playlistId) => {
+        if (!auth.isAuthenticated) {
+            console.log("User is not Authenticated");
+            return;
+        }
+        axios
+            .request({
+                method: "delete",
+                url: `/api/user/playlists/${playlistId}`,
+                headers: { authorization: auth.authToken },
+                data: {},
+            })
+            .then((res) => {
+                console.log(res.data.playlists);
+                setPlaylist([...res.data.playlists]);
+            })
+            .catch((err) => {
+                console.log(err);
+            });
+    };
+
+    const addToPlaylist = (playlistId, video) => {
+        if (!auth.isAuthenticated) {
+            console.log("User is not Authenticated");
+            return;
+        }
+        axios
+            .request({
+                method: "post",
+                url: `/api/user/playlists/${playlistId}`,
+                headers: { authorization: auth.authToken },
+                data: {
+                    video,
+                },
+            })
+            .then((res) => {
+                console.log(res.data.playlist);
+                setPlaylist((s) => [
+                    ...s.map((playlist) =>
+                        playlist._id === res.data.playlist._id
+                            ? res.data.playlist
+                            : playlist
+                    ),
+                ]);
+            })
+            .catch((err) => {
+                console.log(err);
+            });
+    };
+
+    const removeFromPlaylist = (playlistId, video) => {
+        if (!auth.isAuthenticated) {
+            console.log("User is not Authenticated");
+            return;
+        }
+        axios
+            .request({
+                method: "delete",
+                url: `/api/user/playlists/${playlistId}/${video._id}`,
+                headers: { authorization: auth.authToken },
+                data: {
+                },
+            })
+            .then((res) => {
+                console.log(res.data.playlist);
```

Comment:
suggestion: remove `console.log`

Reply:
The console.log files are actually error that I will convert to toasts. So kept it on purpose as reminder while using toasts